### PR TITLE
Run scenario tests in PCS pipeline

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade-services/issues/3901

### CODE REVIEW

File: .vault-config/product-construction-dev.yaml [hunk 1]
```diff
@@ -24,4 +24,4 @@ secrets:
       gitHubBotAccountSecret: 
         location: engkeyvault
         name: BotAccount-dotnet-bot
-      gitHubBotAccountName: dotnet-bot
+      gitHubBotAccountName: dotnet-bot
```
Comment (premun) [inline .vault-config/product-construction-dev.yaml:27]: I'd put the github secret here like it was in your branch

File: azure-pipelines-product-construction-service.yml [hunk 2]
```diff
@@ -133,3 +138,50 @@ stages:
...
+    displayName: Build and Publish Repo
+    pool:
+      name: NetCore1ESPool-Internal
+      demands: ImageOverride -equals 1es-windows-2019
+
+    variables:
+      - name: _BuildConfig
+        value: Release
+      - name: _BuildArgs
+        value: "/p:DotNetSignType=$(_SignType) /p:TeamName=$(_TeamName) /p:DotNetPublishUsingPipelines=$(_PublishUsingPipelines) /p:DotNetArtifactsCategory=$(_DotNetArtifactsCategory) /p:OfficialBuildId=$(BUILD.BUILDNUMBER)"
```
Comment (premun) [inline azure-pipelines-product-construction-service.yml:152]:
```suggestion
        value: >
          /p:DotNetSignType=$(_SignType)
          /p:TeamName=$(_TeamName)
          /p:DotNetPublishUsingPipelines=$(_PublishUsingPipelines)
          /p:DotNetArtifactsCategory=$(_DotNetArtifactsCategory)
          /p:OfficialBuildId=$(BUILD.BUILDNUMBER)"
```

**Review by premun — DISMISSED** [review]

File: eng/templates/jobs/e2e-pcs-tests.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    - name: PcsTestEndpoint
+      value: https://product-construction-int.delightfuldune-c0f01ab0.westus2.azurecontainerapps.io
+    - name: ScenarioTestSubscription
+      value: "Darc: Maestro Staging"
+    - name: MaestroAppId
+      value: $(MaestroStagingAppClientId)
+  steps:
+  - download: current
+    displayName: Download Darc
+    artifact: PackageArtifacts
```
Comment (premun) [inline eng/templates/jobs/e2e-pcs-tests.yml:30]: I think we could filter the files to download darc only?

File: .vault-config/product-construction-dev.yaml (hunk 1 above)
Comment (dkurepa) [reply .vault-config/product-construction-dev.yaml]: Added, not sure why I removed it originally

**Review by premun — APPROVED** [review]

**Pull request merged**